samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -194,7 +216,7 @@ public boolean equals(Object obj) {
   @Override
   public String toString() {
     return "IncomingMessageEnvelope [systemStreamPartition=" + systemStreamPartition + ", offset=" + offset +
-        ", key=" + key + ", message=" + message + ", eventTime=" + eventTime +
-        ", arrivalTime=" + arrivalTime + "]";
+        ", key=" + key + ", message=" + message + ", eventTime=" + eventTimeInstant.toEpochMilli() +
```

Comment:
This will throw an NPE if `eventTimeInstant` is null.

Reply:
defaulted eventTimeInstant to epochMilli(0)